File: problems/p_3/p3.cpp
/*
project euler problem 3
*/
#include <iostream>

bool isPrime(long num){
  if (num == 1) {
    return false;
  }
  long i = 2;
  while (i * i <= num) {
    if (num % i == 0) {
      return false;
    }
    i++;
  }
  return true;
}

long getLargestPrimeFactor(long num){
  long largest = 0;
  long i = 2;
  while (i * i <= num) {
    if (num % i == 0) {
      if (isPrime(i))  {
        largest = i;
      }
    }
    i++;
  }

  return largest;
}

int main(int argc, char const *argv[]) {
  std::cout << "largest prime factor is: " << getLargestPrimeFactor(600851475143) << "\n";
  return 0;
}

File: 3/p3.py
"""
project euler problem 3
"""

def is_prime(num):
    if num == 1:
        return False
    i = 2
    while i*i <= num:
        if num % i == 0:
            return False
        i += 1
    return True

def get_largest_prime_factor(num):
    largest = 0
    i = 2
    while i*i <= num:
        if num%i == 0:
            if is_prime(i):
                largest = i
        i += 1
    return largest

def main():
    print("the largest prime factor is:", get_largest_prime_factor(600851475143))

if __name__ == '__main__':
    main()
